packages/flutter_test/lib/src/binding.dart

```diff
@@ -1109,7 +1109,6 @@ class LiveTestWidgetsFlutterBinding extends TestWidgetsFlutterBinding {
   @override
   void initInstances() {
     super.initInstances();
-    assert(!autoUpdateGoldenFiles);
```

Comment:
I think this assert was added in response to this comment https://github.com/flutter/flutter/pull/17299/files/5dd14e539845a5d9926d00a5d771d7ca7e8bc5ec#diff-0c9f306c093229cac6ca783db4d376d0R116 before StateError is thrown. Not sure whether StateError is actually thrown once assert is removed, whether assert should be moved to different place.

Reply:
Right, but I think the comment is outdated, because we have now tests which run with live widget binding (such as the benchmark test). This test is unique in that it tests the live widget binding explicitly, to ensure benchmarks work correctly.

Alternatively, we might be able to disable that test when `autoUpdateGoldenFiles` is set.

Reply:
If you run in a live widget binding via `flutter run` (with `--update-goldens`), you'll skip the test initialization that sets the golden file comparator, and thus you'll hit this anyway:

https://github.com/flutter/flutter/blob/b27b3d744670bb9312824ec94353b93e7d0424ea/packages/flutter_test/lib/src/goldens.dart#L128

What mechanism are we using to run tests with the live widget binding?  Does it still initialize the golden file comparator?

Reply:
We create a custom binding and simply attach the root widget to it. It does not initialize the golden file comparator, because it shouldn't be necessary for the type of tests which use the live widgets binding.

`packages/flutter/test/scheduler/benchmarks_binding.dart` has the details

Reply:
Perhaps then we need to find another place for this assert?

Reply:
Is seems from Todd's comment that we'll hit an exception on line 128 anyway, so I think it's fine?

Reply:
Conceptually, I think removing the assert entirely is actually the right way to go, since we're moving towards a world where we run widget tests on device using a live widget binding anyway.

But given that, can you also remove this comment?

https://github.com/flutter/flutter/blob/b27b3d744670bb9312824ec94353b93e7d0424ea/packages/flutter_test/lib/src/goldens.dart#L127

Reply:
Done.